src/fides/api/ops/models/privacy_experience.py

```diff
@@ -165,26 +297,302 @@ def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperience:
+            added_experiences.append(privacy_center_experience)
+
+        # See if we need to update an existing Link Overlay to be delivered by Banner instead
+        if banner_delivery_update_needed(overlay_experience, overlay_notices):
+            assert overlay_experience  # For mypy.  Overlay_experience guaranteed to exist here.
+            overlay_experience.update(
+                db=db,
+                data={
+                    "delivery_mechanism": DeliveryMechanism.banner,
+                    "experience_config_id": None,  # We have to unlink the invalid experience config here if it exists
+                    "experience_config_history_id": None,
+                },
+            )
+            updated_experiences.append(overlay_experience)
+
+        # See if we need to create a new overlay Experience for the overlay Notices.
+        if new_experience_needed(
+            existing_experience=overlay_experience, related_notices=overlay_notices
+        ):
+            overlay_experience = PrivacyExperience.create(
+                db=db,
+                data={
+                    "region": region,
+                    "component": ComponentType.overlay,
+                    "delivery_mechanism": DeliveryMechanism.banner,  # Making this banner by default, just to be consistent.
+                },
+            )
+            added_experiences.append(overlay_experience)
+
+    return added_experiences, updated_experiences
+
+
+def config_incompatible_with_region(
+    db: Session,
+    component_type: ComponentType,
+    delivery_mechanism: DeliveryMechanism,
+    region: PrivacyNoticeRegion,
+) -> bool:
+    """Returns True if an ExperienceConfig/PrivacyExperience is invalid for the given region
+
+    Our current primary check is that opt-in and notice only notices have to be displayed in
+    an overlay delivered by a banner.
+    """
+    return (
+        component_type == ComponentType.overlay
+        and delivery_mechanism == DeliveryMechanism.link
+        and bool(
+            get_privacy_notices_by_region_and_component(
+                db, region, ComponentType.overlay  # type: ignore[arg-type]
+            )
+            .filter(
+                PrivacyNotice.consent_mechanism.in_(
+                    [ConsentMechanism.opt_in, ConsentMechanism.notice_only]
+                )
+            )
+            .first()
+        )
+    )
+
+
+def upsert_privacy_experiences_after_config_update(
+    db: Session,
+    experience_config: PrivacyExperienceConfig,
+    regions: List[PrivacyNoticeRegion],
+) -> Tuple[
+    List[PrivacyNoticeRegion], List[PrivacyNoticeRegion], List[PrivacyNoticeRegion]
+]:
+    """
+    Keeps Privacy Experiences in sync with ExperienceConfig changes.
+    Create or update PrivacyExperiences for the regions we're attempting to link to the
+    ExperienceConfig.  If the region cannot be linked, or if it is currently attached
+    and shouldn't be, skip or remove that region.
+    """
```

Comment:
This is another of the primary methods added, to keep PrivacyExperiences updated if PrivacyExperienceConfig is changed.